compiler-core/src/parse/extra.rs

```diff
@@ -15,6 +15,18 @@ impl ModuleExtra {
     pub fn new() -> Self {
         Default::default()
     }
+
+    pub fn contains(&self, offset: u32) -> bool {
```

Comment:
The name "extra contains index" confused me a bit when I was reading it being used. How about something like "is_within_comment"? I think that would make it easier to understand, and it would remove the confusion around which of the things that Extra holds being used here.

Let's also implement this with [a binary search](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.binary_search) to go from `O(n)` to `O(log n)`